controllers/provisioning_controller_test.go

```diff
@@ -144,3 +147,75 @@ func TestProvisioning(t *testing.T) {
 		})
 	}
 }
+
+func TestNetworkStack(t *testing.T) {
+	tests := []struct {
+		name    string
+		ips     []net.IP
+		want    provisioning.NetworkStackType
+		wantErr bool
+	}{
+		{
+			name: "v4 basic",
+			ips:  []net.IP{net.ParseIP("192.168.0.1")},
+			want: provisioning.NetworkStackV4,
+		},
+		{
+			name: "v4 in v6 format: basic",
+			ips:  []net.IP{net.ParseIP("::FFFF:192.168.0.1")},
+			want: provisioning.NetworkStackV4,
```

Comment:
I am suspicious that we got this kind of address in the CI test for https://github.com/openshift/ironic-rhcos-downloader/pull/40#issuecomment-858206600